src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,30 @@
 package mate.academy;
 
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private final Set<EventListener> listeners = ConcurrentHashMap.newKeySet();
+    private final ExecutorService executor = Executors.newCachedThreadPool();
 
+    public void registerListener(EventListener listener) {
+        listeners.add(listener);
     }
 
     public void deregisterListener(EventListener listener) {
-
+        listeners.remove(listener);
     }
 
     public void notifyEvent(Event event) {
-
+        for (EventListener listener : listeners) {
+            CompletableFuture.runAsync(() -> listener.onEvent(event), executor);
+        }
```

Comment:
When using CompletableFuture.runAsync with an ExecutorService, it's important to handle possible exceptions that could be thrown during the execution of the listener's onEvent method. Consider adding exception handling to prevent failing silently.